### DISCUSSION TITLE
How is uBO affected by Web Environment Integrity?

### DISCUSSION THREAD

**Discussion by TheDrawingCoder-Gamer** [uBlockOrigin/uBlock-issues#2751] [Q&A] [answered]
As some of you may know, Google recently proposed [Web Environment Integrity](https://github.com/RupertBenWiser/Web-Environment-Integrity). I want to know how this will affect uBlock Origin, if it makes it to firefox.

**Comment by iam-py-test** [comment]
I don't think this will effect uBo. They explicitly state having browser extensions installed will not effect the attestation<sup>[[1]](https://github.com/RupertBenWiser/Web-Environment-Integrity/blob/main/explainer.md#how-does-this-affect-browser-modifications-and-extensions)[[2]](https://github.com/RupertBenWiser/Web-Environment-Integrity/blob/main/explainer.md#non-goals)</sup>. Of course this is *Google's* word we are going on, so who knows.

**Reply by Yuki2718** [reply, marked as answer]
As far as I understand it, it enables unmodifiable websites so any contents blocker will be affected. This is why monzta posted https://mastodon.social/@gsderp@packmates.org/110741122451704879 on Slack. I don't think Mozilla agrees to this though.

**Reply by iam-py-test** [reply]
I don't recall seeing anything about that in the spec from Google. Where does it enable unmodifiable websites?

**Reply by Yuki2718** [reply]
`https://github.com/RupertBenWiser/Web-Environment-Integrity/blob/main/explainer.md#web-environment-integrity`
A good recap: `https://github.com/RupertBenWiser/Web-Environment-Integrity/issues/125#issuecomment-1645733704`
BTW AG has published an article about it: `https://adguard.com/en/blog/google-api-open-web-adblocking.html`
